src/goto-symex/symex_assign.cpp

```diff
@@ -414,6 +414,25 @@ void goto_symext::symex_assign_symbol(
 
   field_sensitivity.field_assignments(state, lhs_mod);
 
+  // If we just assigned a symbol representing a component of a composite object
+  // (for example, symbol "some_struct.field#1", which represents part of the
+  // symbol "some_struct"), then invalidate "some_struct" in the constant
+  // propagator, as it's now out of date. On future reference we should
+  // reconstruct it from fields using field_sensitivityt::apply.
+  if(ssa_lhs.get_original_expr().id() == ID_member)
+  {
+    ssa_exprt underlying_composite = ssa_lhs;
+    exprt underlying_object = underlying_composite.get_original_expr();
```

Comment:
Can you use a reference wrapper instead, to avoid copies in the loop?

Reply:
This code is gone.